core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -363,6 +363,16 @@ class KafkaApis(val requestChannel: RequestChannel,
     val produceRequest = request.body[ProduceRequest]
     val numBytesAppended = request.header.toStruct.sizeOf + request.bodyAndSize.size
 
+    val transactionalIdAuthorized = if(produceRequest.isTransactional)
+      authorize(request.session, Write, new Resource(ProducerTransactionalId, produceRequest.transactionalId()))
+    else
+      true
+
+    val producerIdAuthorized = if (produceRequest.isIdempotent)
```

Comment:
I think it would be reasonable to say that transactionalId authorization implies producerId authorization. What do you think?